### PR TITLE
fix(docs): Fix anchor tags between pages

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

This PR ensures docs to scroll to the anchor tag on load. 

This doesn't work on our live docs. For example, click on https://ui.docs.amplify.aws/components/authenticator?platform=react#headers--footers and you'll see you're not scrolled to the content.

#### Description of how you validated changes

Validated manually: 

https://user-images.githubusercontent.com/43682783/163648868-87885c26-d19d-4c65-bb8c-4c3767220257.mp4

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]
LGTM

File: docs/src/components/Layout/index.tsx [hunk 1]
```diff
@@ -60,6 +60,20 @@ export default function Page({
...
+    const scrollToHash = () => {
+      const { hash } = window.location;
+
+      if (hash) {
+        document.querySelector(hash)?.scrollIntoView({ behavior: 'smooth' });
+      }
+    };
+    window.addEventListener('load', scrollToHash);
+
+    return () => window.removeEventListener('load', scrollToHash);
```
Comment (reesscot) [inline docs/src/components/Layout/index.tsx:74]: 💯

File: docs/src/components/Layout/index.tsx [hunk 2]
```diff
@@ -60,6 +60,20 @@ export default function Page({
 
     return () => observer.disconnect();
   }, [children]);
+
+  React.useEffect(() => {
+    const scrollToHash = () => {
+      const { hash } = window.location;
+
+      if (hash) {
+        document.querySelector(hash)?.scrollIntoView({ behavior: 'smooth' });
```
Comment (dbanksdesign) [inline docs/src/components/Layout/index.tsx:69]: smooth

**Review by dbanksdesign — APPROVED** [review]
Love it! So Smooth!

**Pull request merged**